editoast/src/generated_data/error/mod.rs

```diff
@@ -146,86 +148,124 @@ fn generate_errors<Ctx: Default>(
 pub fn generate_infra_errors(infra_cache: &InfraCache) -> Vec<InfraError> {
     // Create a graph for topological errors
     let graph = Graph::load(infra_cache);
-
     // Generate the errors
-    let mut infra_errors = generate_errors(
-        ObjectType::TrackSection,
-        infra_cache,
-        &graph,
-        &track_sections::OBJECT_GENERATORS,
-        &[],
-    );
-    infra_errors.extend(generate_errors(
-        ObjectType::Signal,
-        infra_cache,
-        &graph,
-        &signals::OBJECT_GENERATORS,
-        &[],
-    ));
-    infra_errors.extend(generate_errors(
-        ObjectType::SpeedSection,
-        infra_cache,
-        &graph,
-        &speed_sections::OBJECT_GENERATORS,
-        &speed_sections::GLOBAL_GENERATORS,
-    ));
-
-    infra_errors.extend(generate_errors(
-        ObjectType::SwitchType,
-        infra_cache,
-        &graph,
-        &switch_types::OBJECT_GENERATORS,
-        &[],
-    ));
-
-    infra_errors.extend(generate_errors(
-        ObjectType::Detector,
-        infra_cache,
-        &graph,
-        &detectors::OBJECT_GENERATORS,
-        &[],
-    ));
-    infra_errors.extend(generate_errors(
-        ObjectType::BufferStop,
-        infra_cache,
-        &graph,
-        &buffer_stops::OBJECT_GENERATORS,
-        &buffer_stops::GLOBAL_GENERATORS,
-    ));
-
-    infra_errors.extend(generate_errors(
-        ObjectType::OperationalPoint,
-        infra_cache,
-        &graph,
-        &operational_points::OBJECT_GENERATORS,
-        &[],
-    ));
-
-    infra_errors.extend(generate_errors(
-        ObjectType::Route,
-        infra_cache,
-        &graph,
-        &routes::OBJECT_GENERATORS,
-        &routes::GLOBAL_GENERATORS,
-    ));
-
-    infra_errors.extend(generate_errors(
-        ObjectType::Switch,
-        infra_cache,
-        &graph,
-        &switches::OBJECT_GENERATORS,
-        &[],
-    ));
-    infra_errors.extend(generate_errors(
-        ObjectType::Catenary,
-        infra_cache,
-        &graph,
-        &catenaries::OBJECT_GENERATORS,
-        &catenaries::GLOBAL_GENERATORS,
-    ));
-
-    // TODO: generate neutralSections errors
-
+    let infra_errors = Arc::new(Mutex::new(Vec::new()));
+
+    let closures: Vec<Box<dyn Fn() + Send>> = vec![
+        Box::new(|| {
+            let result = generate_errors(
+                ObjectType::TrackSection,
+                infra_cache,
+                &graph,
+                &track_sections::OBJECT_GENERATORS,
+                &[],
+            );
+            let mut infra_errors = infra_errors.lock().unwrap();
```

Comment:
Are we ok to `.unwrap()` here? Are we ok to crash the application in that case? It's indeed possibly an unrecoverable error, but I miss knowledge to evaluate that.